src/coretemp.c

```diff
@@ -0,0 +1,485 @@
+/**
+ * collectd - src/coretemp.c, based on src/protocols.c
+ * Copyright (C) 2016       Slava Polyakov
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
+ *
+ * Authors:
+ *   Slava Polyakov <sigsegv0x0b at gmail.com>
+ **/
+#include "collectd.h"
+#include "common.h"
+#include "plugin.h"
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include <stdio.h>
+#include <sys/types.h>
+#include <dirent.h>
+#include <search.h>
+
+#include "collectd.h"
+#include "common.h"
+#include "plugin.h"
+#include "utils_ignorelist.h"
+
+#if !KERNEL_LINUX
+#error "No applicable input method."
+#endif
+
+#define CORETEMP_PATH "/sys/devices/platform/"
+#define CORETEMP_NAME "coretemp."
+#define CORETEMP_STRLEN (sizeof (CORETEMP_NAME)-1)
+
+static const char *config_keys[] = {
+  "MaxValues",			// [ true ], only, none
+  "ValuesPercentage",		// [ true ], false
+  "ValuesDegrees",		// [ true ], false
+};
+
+// global config vars :)
+int MaxValues = 0;
+int ValuesPercentage = 0;
+int ValuesDegrees = 1;
+
+struct coretemp_core
+{
+  unsigned int tjmax;
+  unsigned int socket;
+  unsigned int core;
+  unsigned int hwmon;
+  int input;
+  char *label;
+};
+
+struct coretemp_core **c = 0;
```

Comment:
I commented on this already in my first review.
There's no need to initialise static variables, but if you insist in doing it anyway, please set pointers to NULL, not 0. Please mark as static while you're at it.